src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -284,4 +295,18 @@ private function hasUserChanged(UserInterface $user)
 
         return false;
     }
+
+    /**
+     * @param string|Role $role
+     *
+     * @return string
+     */
+    private function castRole($role)
+    {
+        if ($role instanceof Role) {
```

Comment:
What about other `RoleInterface` implementations?

Reply:
`RoleInterface` is deprecated and removed in master.

Reply:
The same is true for the `Role` class. But we still need to account for it until they will be dropped in 5.0.